components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/description/query/Query.java

```diff
@@ -213,11 +213,15 @@ private InternalParamCollection extractParams(Map<String, ParamValue> params)
 						QueryTypes.INOUT.equals(queryParam.getType()))) {
 					/* check the exported values */
 					tmpParamValue = exportedParams.get(queryParam.getName());
-					if (tmpParamValue == null && !queryParam.hasDefaultValue()) {
+					if (queryParam.isOptional()) {
+						continue;
+					} else {
+						if (tmpParamValue == null && !queryParam.hasDefaultValue()) {
 						/* still can't find, throw an exception */
-						throw new DataServiceFault(FaultCodes.INCOMPATIBLE_PARAMETERS_ERROR,
-								"Error in 'Query.extractParams', " +
+							throw new DataServiceFault(FaultCodes.INCOMPATIBLE_PARAMETERS_ERROR,
```

Comment:
Formatting issue?

Reply:
No, code indented due to addition of **IF ELSE** on the top